### DISCUSSION TITLE
why exceptions are raised without logged?

### DISCUSSION THREAD

**Discussion by Dongzhenpu** [sqlalchemy/alembic#1008] [Usage Questions] [answered]
I've encountered a problem that my `alembic_version` table's data was deleted, but there are migrations history in `migrations/versions` directory. So when I execute `flask db migrate` there will be an exception which is `Target database is not up to date`, in [api.py](https://github.com/sqlalchemy/alembic/blob/main/alembic/autogenerate/api.py).
But by default, alembic will raise exception without logging it, so what I saw was 
```
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
```
without an ERROR msg, after seen the source code, I'm curious why alembic only raise error without logging it?

**Comment by zzzeek** [comment, marked as answer]
logging of errors using log.error() is intended to be performed when exceptions are caught, not when they are thrown.   that's why Python logging includes an exc_info option; exceptions have a lot more detail when they are caught rather than thrown.   Therefore if the flask-alembic module is using the Alembic API, catching an exception, and then squashing it without it being logged, that's a bug in that application and you should report it over there.

**Reply by Dongzhenpu** [reply]
Many thanks!  @zzzeek 
I've found it's my mistake that I didn't use the latest version of flask_migrate, the problem was the same as here [miguelgrinberg/Flask-Migrate#384](https://github.com/miguelgrinberg/Flask-Migrate/issues/384).
The old version has a bug as this [comment](https://github.com/miguelgrinberg/Flask-Migrate/issues/384#issuecomment-777406725).
Thanks again for your explanation, make me I understand why exceptions are thrown without being logged here.
